### ISSUE TITLE
Wrong Color using PiCapture and opencv

### ISSUE THREAD

**Issue by 541380000** [orgicus/PiCapture#2]
Dear author:
I am using PiCapture and I met with some problem.
I run
raspistill -o image.jpg
in the Terminal of Raspberry Pi, and get a good picture.
But when I use the PiCapture simply capture a Mat and use imshow to show it .But the video is blue, All the red color disappeared. I tried the color_callback() in the README.MD, but it cannot be compiled, and the error is : "image" is not decleared in the scope.
Could you please help me with that? My e-mail is 1327804001@qq.com.

**Comment by 541380000** [comment]
It seems that the openCV use BGR but the raspberry Pi use RGB.
So the Blue and red is opposite,and the green is ok

**Comment by 541380000** [comment]
Solved Again!!!!!!!!

in the PiCapture, the mmal code is 

format->encoding = MMAL_ENCODING_RGB24;
format->encoding_variant = MMAL_ENCODING_RGB24;

and i just  refer to the API of MMAL and get this 

`default_formats = (MMAL_ENCODING_OPAQUE, MMAL_ENCODING_I420, MMAL_ENCODING_RGB24, MMAL_ENCODING_BGR24, MMAL_ENCODING_RGBA, MMAL_ENCODING_BGRA)`

and then I change RGB24 to BGR24 and the video turned out to be ok!

**Comment by orgicus** [comment]
@541380000 Sorry, I got confused: I thought you have already fixed the issue running `sudo rpi-update`.  Please let me know what version of Raspberry Pi, Pi Camera module and Raspian you are using. 

As a quick workaround, you have two options:

1. Use OpenCV for the conversion: `cv::cvtColor(im, im, RGB2BGR);` (bare in mind this will cost a bit of processing power)
2. Change these two lines ([84-85](https://github.com/orgicus/PiCapture/blob/master/src/PiCapture.cpp#L84)) in PiCapture.cpp:

from:
```c++
format->encoding = MMAL_ENCODING_RGB24;
        format->encoding_variant = MMAL_ENCODING_RGB24;
```
to:
```c++
format->encoding = MMAL_ENCODING_BGR24;
        format->encoding_variant = MMAL_ENCODING_BGR24;
```

Use what is easiest for you

**Comment by 541380000** [comment]
Thanks so much for your help and encouragement. I have solved this problem by changing the RGB24 to BGR24.Additionally, I am using Raspberry Pi 3B now. I run the PiCapture quiet well on raspberry Pi 2 ,but because it stopped working for some reason. So I hurriedly use Pi3.

I have communicated with you via email for two times. you are not only talanted but helpful and responsable. thanks so much for your help.

It has been 2:00am in China and I gonna go to sleep.

Good noon!

**Comment by 541380000** [comment]
Thanks so much for your help and encouragement. I have solved this problem by changing the RGB24 to BGR24.Additionally, I am using Raspberry Pi 3B now. I run the PiCapture quiet well on raspberry Pi 2 ,but because it stopped working for some reason. So I hurriedly use Pi3.

I have communicated with you via email for two times. you are not only talanted but helpful and responsable. thanks so much for your help.

It has been 2:00am in China and I gonna go to sleep.

Good noon!
